kedro/framework/cli/starters.py

```diff
@@ -524,8 +549,8 @@ def _create_project(template_path: str, cookiecutter_args: dict[str, Any]):
     )
     add_ons = extra_context.get("add_ons")
 
-    # Only non-starter projects have configurable add-ons
-    if template_path == str(TEMPLATE_PATH):
+    # Only core template and spaceflights-pyspark have configurable add-ons
+    if template_path == str(TEMPLATE_PATH) or add_ons == "6":
```

Comment:
In `fetch_template_based_on_add_ons` there's the check `if add_ons and "Pyspark" in add_ons:` and here we check that `add_ons == "6"`. It seems that in both cases the add_ons comes from `add_ons = extra_context.get("add_ons")`, so could we make the checks consistent and either check for 6 or pyspark?

Reply:
And shouldn't this check if `6` (or pyspark) is in the selected rather than an exact match?

Reply:
@noklam's refactor made use of the `_parse_add_ons_input` in `_make_cookiecutter_args` so we have the selected add-ons in the cookiecutter context.

Reply:
oops sorry misread this, yes it shouldn't be an exact match.